lib/src/model/package_graph.dart

```diff
@@ -1029,3 +1070,41 @@ class InheritableElementsKey {
 
   InheritableElementsKey(this.element, this.library);
 }
+
+extension on Comment {
+  /// A mapping of all comment references to their various data.
+  Map<String, CommentReferenceData> get data {
+    if (references.isEmpty) return const {};
+
+    var data = <String, CommentReferenceData>{};
+    for (var reference in references) {
+      var commentReferable = reference.expression;
+      String name;
+      Element? staticElement;
+      if (commentReferable case PropertyAccessImpl(:var propertyName)) {
+        var target = commentReferable.target;
+        if (target is! PrefixedIdentifierImpl) continue;
+        name = '${target.name}.${propertyName.name}';
+        staticElement = propertyName.staticElement;
+      } else if (commentReferable case PrefixedIdentifier(:var identifier)) {
+        name = commentReferable.name;
+        staticElement = identifier.staticElement;
+      } else if (commentReferable case SimpleIdentifier()) {
+        name = commentReferable.name;
+        staticElement = commentReferable.staticElement;
+      } else {
+        continue;
+      }
+
+      if (staticElement != null && !data.containsKey(name)) {
+        data[name] = CommentReferenceData(
```

Comment:
I haven't thought about this deeply like I'm sure you have, but at first glance this appears to be prone to having information overwritten if it's possible to have two comment references with the same `name` that resolve to different elements.

For example, could a reference to a parameter in one method and a parameter with the same name in a different method cause the map to tell us that both references refer to the second method's parameter? Or a similar question about references to type parameters?

Reply:
Good question! Since this `get data` is made available on  a single Comment, each reference in square brackets with the same name must refer to the same thing. Like in

```
/// Text [foo] and [bar] and [foo].
```

There is never a change to the scope within a doc comment, so `[foo]` and `[foo]` always refer to the same element.

And then the Map of CommentReferenceData is not merged with any other maps from any other comments. It is just stored in a ModelNode for the documented element.